src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsEnumMemberExpression.cs

```diff
@@ -54,15 +54,31 @@ public IEnumerable<EdmError> Errors
         private IEnumerable<IEdmEnumMember> ComputeReferenced()
         {
             IEnumerable<IEdmEnumMember> member;
+            if (this.expression.EnumType != null)

```

Comment:
>expression.EnumType != null [](start = 21, length = 27)

So you're basing whether to use json or xml parser based on whether or not you have a type, which is a bit subtle -- we should add a comment explaining the logic here.

If we think the logic for parsing the json enum member is generic, we could rename the method to something like TryParseUnqualifiedEnumMember (and add a comment to that routine that JSON doesn't type the enum members)

Reply:
Added the comments.

---
In reply to: [442942810](https://github.com/OData/odata.net/pull/1775#discussion_r442942810) [](ancestors = 442942810)